### ISSUE TITLE
Error on creating resource: "Error: couldn't find resource (21 retries)"

### ISSUE THREAD

**Issue by mgoldman-solarwinds** [terraform-aws-modules/terraform-aws-rds#128]
Bear with me as I'm newer to terraform, please. I've copied the working example of mysql for 2.0.0, using terraform 0.12. When I go to create the resource, it times out: 

```
module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: Creating...
module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: Still creating... [10s elapsed]
...
module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: Still creating... [3m40s elapsed]
module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: Still creating... [3m50s elapsed]

Error: couldn't find resource (21 retries)

  on .terraform/modules/aws-rds.db/terraform-aws-modules-terraform-aws-rds-fedd420/modules/db_instance/main.tf line 39, in resource "aws_db_instance" "this":
  39: resource "aws_db_instance" "this" {

```

I'm not sure what I'm doing wrong, because the resource is being created in AWS, but it's still in its spinning up "modifying" stage, so of course it isn't available yet. Maybe I'm missing something on how to change the timeout on terraform to be more patient? 

Here is my `main.tf`:

```
module "db" {
  source  = "terraform-aws-modules/rds/aws"
  version = "2.0.0"


  identifier = myrdsinstance

  # All available versions: http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_MySQL.html#MySQL.Concepts.VersionMgmt
  engine            = "mysql5.6"
  engine_version    = "5.6.41"
  instance_class    = db.m5.2xlarge
  allocated_storage = 300
  storage_encrypted = false
  name     = dbname
  username = username
  password = totallysecurepassword
  port     = 3306

  vpc_security_group_ids = sg-1234

  maintenance_window = var.maintenance_window #default
  backup_window      = var.backup_window           #default

  multi_az = true

  # disable backups to create DB faster
  backup_retention_period = 0

  tags = {
    "foo"       = "bar'
  }

  # enabled_cloudwatch_logs_exports = ["audit", "general"]

  # DB subnet group
  subnet_ids = "${var.subnet_id}"

  # DB parameter group
  family = "mysql5.6"

  # DB option group
  major_engine_version = "5.6"

  # Snapshot name upon DB deletion
  final_snapshot_identifier = "${var.terr_name}_final_snapshot"

  # Database Deletion Protection
  deletion_protection = true

  parameters = [
    {
      name  = "character_set_client"
      value = "utf8"
    },
    {
      name  = "character_set_server"
      value = "utf8"
    }
  ]

  options = [
    {
      option_name = "MARIADB_AUDIT_PLUGIN"

      option_settings = [
        {
          name  = "SERVER_AUDIT_EVENTS"
          value = "CONNECT"
        },
        {
          name  = "SERVER_AUDIT_FILE_ROTATIONS"
          value = "37"
        },
      ]
    },
  ]
}
```

**Comment by antonbabenko** [comment]
Looks good in general, but maybe you are trying to allocate 300GB and it takes some time?

You can try to increase the timeout by setting:
```hcl
timeouts = {
  create = "60m"
  delete = "60m"
  update = "60m"
}
```

**Comment by mgoldman-solarwinds** [comment]
Thanks for responding. I tried both making a smaller storage (50G) and using the timeout provided, but it still fails after 21 retries. It's very odd. What else can I get you to help figure this out?

**Comment by antonbabenko** [comment]
"21 retries" sound magical :) Can it be that your AWS credentials expire every 4 minutes or so?

I recommend opening an issue in [terraform-provider-aws](https://github.com/terraform-providers/terraform-provider-aws) because it is unlikely to be a problem in this module itself.

Make sure to attach a log:
```
export TF_LOG=debug
terraform apply
```

**Comment by mgoldman-solarwinds** [comment]
Some interesting snippets: 

```
2019-06-13T14:16:27.886-0400 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/13 14:16:27 [INFO] Waiting for DB Instance (UT) to be available
2019-06-13T14:16:27.886-0400 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: 2019/06/13 14:16:27 [DEBUG] Waiting for state to become: [available storage-optimization]
module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: Still creating... [10s elapsed]
```

```
for "module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]"
2019/06/13 14:18:45 [TRACE] dag/walk: vertex "module.aws-rds.output.this_db_instance_hosted_zone_id" is waiting for "module.aws-rds.module.db.output.this_db_instance_hosted_zone_id"
2019/06/13 14:18:45 [TRACE] dag/walk: vertex "module.aws-rds.module.db.output.this_db_instance_address" is waiting for "module.aws-rds.module.db.module.db_instance.output.this_db_instance_address"
2019/06/13 14:18:45 [TRACE] dag/walk: vertex "module.aws-rds.module.db.module.db_instance.local.this_db_instance_status" is waiting for "module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]"
```

```
2019-06-13T14:20:22.225-0400 [DEBUG] plugin.terraform-provider-aws_v2.14.0_x4: </DescribeDBInstancesResponse>
2019/06/13 14:20:22 [DEBUG] module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]: apply errored, but we're indicating that via the Error pointer rather than returning it: couldn't find resource (21 retries)
2019/06/13 14:20:22 [TRACE] module.aws-rds.module.db.module.db_instance: eval: *terraform.EvalMaybeTainted
2019/06/13 14:20:22 [TRACE] EvalMaybeTainted: module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0] encountered an error during creation, so it is now marked as tainted
2019/06/13 14:20:22 [TRACE] module.aws-rds.module.db.module.db_instance: eval: *terraform.EvalWriteState
2019/06/13 14:20:22 [TRACE] EvalWriteState: writing current state object for module.aws-rds.module.db.module.db_instance.aws_db_instance.this[0]
```

Finally ends with this, and I think you might be right about terraform-provider-aws:

```
Error: couldn't find resource (21 retries)

  on .terraform/modules/aws-rds.db/terraform-aws-modules-terraform-aws-rds-fedd420/modules/db_instance/main.tf line 39, in resource "aws_db_instance" "this":
  39: resource "aws_db_instance" "this" {
```

**Comment by mgoldman-solarwinds** [comment]
Looks like this was solved by upgrading from 0.12.1 -> 0.12.2. Closing out.
